Sources/PhishingDetection/PhishingDetectionService.swift

```diff
@@ -0,0 +1,181 @@
+//
+//  PhishingDetectionService.swift
+//
+//  Copyright © 2023 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import CryptoKit
+import Common
+
+public struct HashPrefixResponse: Decodable, Encodable {
+    public var hashPrefixes: [String]
+    public var revision: Int
+}
+
+public struct FilterSetResponse: Decodable, Encodable {
+    public var filters: [Filter]
+    public var revision: Int
+}
+
+public struct Filter: Decodable, Encodable {
+    public var hashValue: String
+    public var regex: String
+
+    enum CodingKeys: String, CodingKey {
+        case hashValue = "hash"
+        case regex
+    }
+
+    public init(hashValue: String, regex: String) {
+        self.hashValue = hashValue
+        self.regex = regex
+    }
+}
+
+public struct MatchResponse: Decodable, Encodable {
+    public var matches: [Match]
+}
+
+public struct Match: Decodable, Encodable {
+    var hostname: String
+    var url: String
+    var regex: String
+    var hash: String
+
+    public init(hostname: String, url: String, regex: String, hash: String) {
+        self.hostname = hostname
+        self.url = url
+        self.regex = regex
+        self.hash = hash
+    }
+}
+
+public protocol PhishingDetectionServiceProtocol {
+    var filterSet: [Filter] {get set}
+    var hashPrefixes: [String] {get set}
+    func isMalicious(url: URL) async -> Bool
+    func updateFilterSet() async
+    func updateHashPrefixes() async
+    func getMatches(hashPrefix: String) async -> [Match]
```

Comment:
is this used anywhere?

Reply:
Only for mocking the service when it's passed to PhishingDetectionDataActivities to test it. Might be overkill tbh we could probably remove the protocol.